### PR TITLE
Missing from #5433, automs order 512, and nilpotency test for permgroups

### PR DESCRIPTION
**Pull request by hulpke**
When translation #5394 to #5433, the correction for units for nonassociative rings, and a basic division routine were not added.
This is adding these two changes (which I hope are uncontroversial).

I'm also adding a minor strategy change to automorphism group computation that affects group of order 512 (and was a forum complaint) -- basically the assumption of relying on the small groups library does not work here, as no identification function exists.

Also corrected is an issue that niloptency testing in large degree permutation groups might run into a recursion trap (support email 4/26/23)

Finally there was a typo in the presentation for `WeylGroupFp("E",8)` that has been fixed.

### CODE REVIEW

File: lib/ringsc.gi [hunk 1]
```diff
@@ -918,27 +918,27 @@ InstallOtherMethod(IsUnit,"for finite Rings",
...
-      a:=a*e;
-      if a=o then
-        # power is one. So previous power is inverse
-        return true;
-      fi;
-    until a in pow;
-    # repeats without hitting the one element, cannot be a unit
-    return false;
+
+  if not IsAssociative(R) then
```
Comment (fingolfin) [inline lib/ringsc.gi:925]: What is the difference herr?

Comment (hulpke) [reply lib/ringsc.gi:925]: The difference is that the new code only runs if the ring is associative (it is probably not correct in the non-associative case).

Comment (fingolfin) [reply lib/ringsc.gi:925]: I don't understand? The code you replace also differentiates between associative and non-associative. E.g. there is a check `if IsAssociative(R) and One(R) <> fail then`

Comment (hulpke) [reply lib/ringsc.gi]: I had noted that the code for ring.gi and ringsc.gi differed from my version. The one for ring.gi was definitively wrong. Thus I replaced both. 
I'll take the ringsc.gi out.

**Review by fingolfin — APPROVED** [review]
Looks good to me, thanks

**Pull request merged**